### PR TITLE
central command

### PR DESCRIPTION
**Pull request by zimbatm**
This is just a demo of another direction this project could take.

I wanted to avoid having to define two different files, one for the bin and one for the nixos config, and having to agree on the attribute path to export.

With this approach, each target definition is only a single nix file. There is a central `nixos-build` command that can be invoked.

The command respects the same standard to select the configuration.nix so it should be more portable.

Eg:
```
# build the current system configuration as a google compute image.
$ nixos-build --target gce
# provide the configuration and create a virtualbox image out of it
$ nixos-build --configuration ./config.nix --target virtualbox
```

### CODE REVIEW

**Comment by Lassulus** [comment]
This looks nice, I will have a more thorough look later.
Just 2 usecases I want to highlight:
`dd if=$(nixos-build --target iso) of=/dev/sda`

and I want users to be able to just copy one of the generators for their own project (more or less easily), for example the iso generator.

The first usecase is way more important than the first.

**Comment by zimbatm** [comment]
The first use-case should be handled properly if the outputs contain only one file right now. We might have to add a targetGlob attribute otherwise.

Assuming that the user would install nixos-generator, the second use-case should be supported as well. It shouldn't be too hard to add an extension point in case they want to provide custom formats.

PS: I want to rename `target` to `format` and `nixos-build` to `nixos-generator` as it's a more accurate names
PPS: the file locations are not quite right now, files can be moved around

File: nixos-generate [hunk 1]
```diff
@@ -3,7 +3,7 @@ set -euo pipefail
 
 ## Configuration
 
-libexec_dir=$(dirname "$0")
+libexec_dir="${0%/*}"
```
Comment (Lassulus) [inline nixos-generate]: wuhuu, nice trick. Thats the kind of magic I like ;D

Comment (zimbatm) [reply nixos-generate]: hehe :)

**Comment by Lassulus** [comment]
looks really cool.
kexec-bundle seems broken. apart from that, this should be ready to merge? or do you want to change more?

**Comment by zimbatm** [comment]
thanks!

**Pull request merged**